### ISSUE TITLE
Easier way to use Data Processing steps outside of datamodule

### ISSUE THREAD

**Issue by nilsleh** [microsoft/torchgeo#1780]
### Summary

Normalization and Augmentations are defined in the `on_after_batch_transfer()` function of the Datamodules to compute them on GPU like recommended from lightning. However, a downside of this is that you always have to pass the datamodule ino `.fit` and `.test`. Especially, for the latter, it can be convenient to test on separate dataloaders, however, those are then just "raw" dataloaders without normalization etc. being applied. Took me a minute to find that this was the reason for the funky test results. Currently, I am writing a custom `collate_fn` and set it to the dataloader that I am getting from a datamodule, however, it would be nice if this could be handled more easily. Open to hear thoughts about this, or suggestions for an easier ways to handle this than what I am doing at the moment.

**Comment by adamjstewart** [comment]
I can understand why you would want to be able to use a dataset if a data module doesn't exist, but why would you want to use a dataset if a data module does exist?

**Comment by nilsleh** [comment]
In order to do `trainer.validate(model, dataloaders=datamodule.val_dataloader())` but not having to implement my own normalization scheme as a collate fn for every dataloader from a datamodule I want to use. So for example say I train one model and want to validate it on a bunch of datasets, then I could pass multiple dataloaders from different datasets or datamodules to `trainer.validate()`

**Comment by adamjstewart** [comment]
But why not use `trainer.validate(model, datamodule=datamodule)` for all data modules?

**Comment by nilsleh** [comment]
If you pass a datamodule, it will only select the predefined validation loader and validate on that, but maybe I would like to validate on the train set and the validation set, for example when taking a pre-trained model and checking performance without training.

**Comment by robmarkcole** [comment]
I think at a minimum we should improve the documentation to state when augmentations are and are not applied. For example I assumed they are performed in the dataset get_item, but they are not/ 

I am having extra headaches as I swapped the torchgeo implementation of `AugmentationSequential` for the kornia implementation, and now I am having to update my dataset to return masks as float not long, and make sure the channel dim is/is not present at different points, it is not fun :-)

**Comment by isaaccorley** [comment]
Just want to clarify that _for the majority of datamodules there are no augmentations applied_, only a normalization of the images. We try not to prescribe which augmentations should be used for what dataset as this should be left to the user. 

There are 2 options:
- override the datamodule and the train/val/test/predict aug attributes to define your own Kornia AugmentationSequential pipeline (this performs batched GPU augmentations)
- override the datamodule and its setup method to pass in your own transforms to the train/val/test datasets (this performs augs at the getitem level for per sample augmentation). This option will work using the dataloader e.g. datamodule.train_dataloader() without passing the datamodule to trainer.fit/test

**Comment by robmarkcole** [comment]
Thanks for the clarification - it  makes sense that there are basic augmentations to always apply during training and which we don't need to inspect (ie normalisation) and others to experiment with and visualised for sanity checking. Therefore is it a reasonable workflow to pass the later kind to the dataset in setup, and still have the former applied at the data module level?
